### DISCUSSION TITLE
Cache ttl tuning for infrequent runs (2 hours between run)

### DISCUSSION THREAD

**Discussion by zeldigas** [renovatebot/renovate#24214] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.44.0

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

Placeholder value, please select the correct response from the dropdown

### Wanted end result.

My renovate setup is running on schedule - every 2 hours. Runs are done on CI system and there is no guarantee that subsequent scheduled runs will be executed on same host.

I'd like to reuse cache between runs to avoid requesting all info about dependencies again and again. Ideally - to make sure that cache is really used by next renovatebot run.

The thing that confuses me - is default ttl value specified in https://docs.renovatebot.com/self-hosted-configuration/#cachettloverride. It says that it is equal to one hour.

If I got it correctly, then I assume that all this save/restore operation for cache directory might be useless, because of cache will be considered stale.

So the question is - do I need to tune configuration somehow to use case efficiently or it will just work with default settings in described setup?

### What you tried so far.

So far I've configured archiving `$cacheDir/renovate` directory. Before next run, I'm downloading archive and unpackaging it - it works (when 2 runs are run one by another on different machines).

**Comment by rarkins** [comment]
Are you ok to live with results which are >1 hour out of date though?

**Reply by zeldigas** [reply]
Good point, I spent some time looking into how renovate works with cache and see that different date is cached differently. - list of releases for short period of time, and info about individual versions for longer period. So probably it's not an issue.

The only thing that I don't get so far - time to cache release notes/changelog. In subsequent runs with cache I see that changelog are fetched from github. I don't get why it happens so far. As changelogs are usually not modified (when created), not sure why I see repeated request.

**Reply by rarkins** [reply, marked as answer]
I forget the details but changelog cache period does use a back off algorithm of sorts depending on the release date of each package if known. If notes change they usually change within the first day, occasionally first week, and almost never after that. You're right that they change much mess often
